tracer/test/Datadog.Trace.Tests/Snapshots/PublicApiTests.PublicApiHasNotChanged.verified.txt

```diff
@@ -295,6 +295,13 @@ namespace Datadog.Trace
         public const string SqlRows = "sql.rows";
         public const string Version = "version";
     }
+    [System.AttributeUsage(System.AttributeTargets.Method | System.AttributeTargets.All, AllowMultiple=false)]
+    public class TraceAttribute : System.Attribute
+    {
+        public TraceAttribute() { }
+        public string OperationName { get; set; }
+        public string ResourceName { get; set; }
+    }
```

Comment:
This ship may have sailed, but just wondering&hellip; 

Do we definitely want to include this attribute in Datadog.Trace? We _could_ have this in a tiny separate library. That way, users don't _have_ to use manual instrumentation, and we absolve them of any requirement to keep versions in sync etc.

I have a vague recollection that @kevingosse mentioned the proposed "do not trace" marker similarly being in a separate PR, but I can't quite remember if/how that would work, so it may have been something else. 

Obviously this lib would need to be _very_ slim, devoid of any logic, and (mostly) immutable!

Reply:
OTOH, I think users can already define their own attribute, as long as it's called `Datadog.Trace.TraceAttribute`.

Reply:
I prefer not having attributes at all 😂 as @lucaspimentel said in the latest message. Just like  `IgnoresAccessChecksToAttribute`. That way we can extend the support for others well-known attributes for tracing using the same implementation. 😛

Reply:
We can make the feature work just based on type name. I'd be very happy with either 1) Having this in a separate, slim library or 2) Having the user define the attribute. There's some level of friction for the user to set up the attribute themselves because customization will require specific property names, so maybe we should do 1) and customers can roll their own 2) if they want?

Reply:
> I prefer not having attributes at all 😂 as @lucaspimentel said in the latest message. Just like `IgnoresAccessChecksToAttribute`. That way we can extend the support for others well-known attributes for tracing using the same implementation. 😛

I already plan to do a type name comparison to try and read New Relic's TraceAttribute, and othes 😄

Reply:
> There's some level of friction for the user to set up the attribute themselves because customization will require specific property names, so maybe we should do 1) and customers can roll their own 2) if they want?

I agree: `Datadog.Trace.Annotations`?

Reply:
All these suggestions were implemented in the latest changes 😄